From 609c53f974caee6b75c6ffa259cf6fcda671e39b Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Sun, 25 Feb 2018 22:25:17 +0200
Subject: [PATCH] Revert "Allow underscores that are part of tab titles to be
 shown in the tabs menu"

This reverts commit 300d136e1c9ab2c2a687c11a8ccd7b470816ee36.
---
 terminal/terminal-window.c | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/terminal/terminal-window.c b/terminal/terminal-window.c
index e678c8f7..4af3bc64 100644
--- a/terminal/terminal-window.c
+++ b/terminal/terminal-window.c
@@ -2956,7 +2956,7 @@ terminal_window_rebuild_tabs_menu (TerminalWindow *window)
   GtkWidget      *page;
   GSList         *group = NULL;
   GtkRadioAction *radio_action;
-  gchar           name[50], menu_item_name[100];
+  gchar           name[50];
   GSList         *lp;
   GtkAccelKey     key = {0};
 
@@ -3011,26 +3011,22 @@ G_GNUC_END_IGNORE_DEPRECATIONS
       g_object_set_qdata_full (G_OBJECT (page), tabs_menu_action_quark,
                                radio_action, g_object_unref);
 
-G_GNUC_BEGIN_IGNORE_DEPRECATIONS
       /* add action in the menu */
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
       gtk_ui_manager_add_ui (window->priv->ui_manager, window->priv->tabs_menu_merge_id,
                              "/main-menu/tabs-menu/placeholder-tab-items",
                              name, name, GTK_UI_MANAGER_MENUITEM, FALSE);
-      /* allow underscore to be shown */
-      g_snprintf (menu_item_name, sizeof (menu_item_name), "/main-menu/tabs-menu/placeholder-tab-items/%s", name);
-      gtk_menu_item_set_use_underline (GTK_MENU_ITEM (gtk_ui_manager_get_widget (window->priv->ui_manager, menu_item_name)), FALSE);
+G_GNUC_END_IGNORE_DEPRECATIONS
 
       if (npages > 1)
         {
           /* add to right-click tab menu */
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
           gtk_ui_manager_add_ui (window->priv->ui_manager, window->priv->tabs_menu_merge_id,
                                  "/tab-menu/tabs-menu/placeholder-tab-items",
                                  name, name, GTK_UI_MANAGER_MENUITEM, FALSE);
-          /* allow underscore to be shown */
-          g_snprintf (menu_item_name, sizeof (menu_item_name), "/tab-menu/tabs-menu/placeholder-tab-items/%s", name);
-          gtk_menu_item_set_use_underline (GTK_MENU_ITEM (gtk_ui_manager_get_widget (window->priv->ui_manager, menu_item_name)), FALSE);
-        }
 G_GNUC_END_IGNORE_DEPRECATIONS
+        }
 
       /* set an accelerator path */
       g_snprintf (name, sizeof (name), "<Actions>/terminal-window/goto-tab-%d", n + 1);
-- 
2.16.2

